src/foam/dao/history/HistoryDAO.js

```diff
@@ -128,34 +129,46 @@ foam.CLASS({
     {
       name: 'put_',
       javaCode: `
-        Subject subject = (Subject) x.get("subject");
-        User user = subject.getUser();
-        User agent = subject.getRealUser();
-        FObject current = this.find_(x, obj);
-    
+      Subject subject = (Subject) x.get("subject");
+      User user = subject.getUser();
+      User agent = subject.getRealUser();
+      FObject current = this.find_(x, obj);
+      Object objectId = ((PropertyInfo) obj.getClassInfo().getAxiomByName("id")).f(obj);
+      boolean isCreate = objectId == null || 
+                          current == null ||
```

Comment:
Why are the other checks needed? Isn't 'current == null' sufficient?

Reply:
We were both  fighting with this. I have something similar in serviceproviderawaredao. 
It suggest bad data - we have a 0 entry.  reviewing again.

Reply:
Yes, make sense. If there is no data find from mdao, it should be "create".

Reply:
Looking at notes, it was an optimization to avoid the find, then the order was changed, and in the end we both need the obj object regardless, so can be removed.

Reply:
@kgrgreer We can't. Because when the obj is "create" (first time put), there is no Id in obj (in medusa). So, it can't find anything.

Reply:
We have different easydao setup in development and medusa. "current == null" is used to make sure development working correctly. and the rest of checks is used to make sure medusa currectly.